### PR TITLE
Add reference probabilistic forecast

### LINKED ISSUE #195: probabilistic forecasts from observations
some options include:

* persistence ensemble (most recent N observations at same time of day)
* raw persistence ensemble (most recent N observations)
* complete history persistence ensemble (all observations at same time of day)
* climatology (all observations)

### PR DESCRIPTION
**Pull request by dplarson**
- [x] Closes #195 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

The goal of this PR is to add a reference probabilistic forecast method. My initial idea is to implement the Persistence Ensemble (PeEn) method [1], which is fairly commonly used as a baseline probabilistic forecast in the literature and relies solely on historical observations to estimate an empirical CDF, which is then propagated into future. Similar to (dull) persistence, this works well in short horizons and during non-changing conditions (e.g. extended periods of clear-sky or overcast weather). Also, we can implement variations of PeEn such as using clear-sky index observations (similar to "Smart Persistence") as well as matching the empirical CDF by hour of the day, which in turn would provide a similar effect as using clear-sky index but for data without clear-sky models (e.g. net load). [2]

Overall the method is:
1. estimate the empirical CDF from historical observations for `data_start` to `data_end`
2. for each probability interval (e.g. 25%, 50%, 75%), forecast the future probabilities for `forecast_start` to `forecast_end` as the empirical CDF (`ECDF`) (e.g `fx(25%) = ECDF(25%)`, `fx(50%) = ECDF(50%)`, etc.)

[1] Alessandrini et al. (2015) "An analog ensemble for short-term probabilistic solar power forecast", Applied Energy
[2] Dazhi Yang (2019) "A universal benchmarking method for probabilistic solar irradiance forecasting", Solar Energy

### CODE REVIEW

**Comment by dplarson** [comment]
Current areas that required feedback:
- [ ] function name: `persistence_probabilistic`, `probabilistic_persistence`, `persistence_distribution`, other???
   - I am leaning towards `persistence_probabilistic` in order to match the deterministic persistence naming convention (`persistence_scalar`, `persistence_interval`, etc.)
- [ ] scope: only implement base PeEn method or also modification with clear-sky index?
   - I am leaning towards only implementing base PeEn to start (with a follow-up issue/PR if we decide to expand)
- [ ] `data_start, data_end`: keep general (as is) or automatically match by hour of the day (e.g. forecast March 4th at 12:00–13:00 at 5-minute intervals using (a) data from March 3rd at 12:00–13:00 or (b) data from March 3rd 12:00–13:00, March 2nd 12:00–13:00, ..., April 29th 12:00–13:00
   - I am leaning towards (a) keep general as user-defined `data_start, data_end` with no match by hour of day

**Comment by dplarson** [comment]
Also, the code in this PR should be treated as pseudo-code to illustrate the main ideas. (It might work as is, but I haven't tested/confirmed its functionality).

**Comment by dplarson** [comment]
@wholmgren no rush, but I'm hoping to get your feedback on whether this draft PR aligns with your thoughts on a reference probabilistic forecast. Once we agree on the general approach, I can work through the code.

**Comment by wholmgren** [comment]
@dplarson thanks for the draft and I agree this is a good point to discuss and think harder.

> variations of PeEn such as using clear-sky index observations (similar to "Smart Persistence")

I'd say out of scope for DOE funding because the following is more general, used in literature, and gets at nearly the same result for many users

> matching the empirical CDF by hour of the day, which in turn would provide a similar effect as using clear-sky index but for data without clear-sky models (e.g. net load). [2]

I think this is important to get right in the initial design. Are you thinking of an entirely different function for this or adding something like a `groupby` kwarg to the currently proposed function? I don't know the right answer. My initial guess is a `groupby` kwarg could be straightforward, perhaps with a special case `None` for pure persistence.

What about for forecasts that have subhourly resolution? Should we construct the CDF from the subhourly window over the last n days? I think that should be allowed. 

Mild preference for `persistence_probabilistic` over `persistence_distribution`.

> data_start, data_end: keep general (as is) or automatically match by hour of the day 

If I understand correctly, the addition of the `groupby` kwarg or a separate function would allow for both of these options.

For matching by hour of day (or other interval), perhaps this base function should use all possible data from `data_start` to `data_end`, while the higher level calls in `reference_forecasts.main` provide more structure (e.g. only use the last 20 days).

**Comment by dplarson** [comment]
Thanks for the feedback. My initial thoughts are:
1. have two functions: one function constructs the CDF from `data_start` to `data_end` (with no correction for time of day) while the other constructs the CDF using data from the last `n` days matched by some sort of `groupby` operation (e.g. to predict 10am, use the CDF from the 10am observations of the past 20 days)
2. for the function that matches by time of day, have the user provide the number of past days to consider and then have the function (try to) automatically match the CDFs by time of day

For item #2, it would be something like:
```python
# forecast timestamps
fx_index = pd.date_range(start=fx_start, end=fx_end, freq=interval_length, closed=closed)

# load observations from past 7 days (at the original observation freq)
data_start = data_end - pd.Timedelta("7D")
obs = load_data(data_start, data_end)

# resample observations to match the forecast freq (e.g. 5-minute to 1-hour)
obs = resample(obs)

# compute time of day [-] (e.g. 12:00=0.00, 12:15=0.25, ..., 23:45=23.75)
timeofday = fx_index.hour + fx_index.minute / 60.0

# for each time of day, construct a CDF from observations during that same time of day
for t in timeofday:
    F = ECDF(obs[obs["timeofday"] == t])    # construct CDF
    fx[fx["timeofday"] == t] = F(50%)            # apply CDF

```

Definitely some details to work out, but I think what I've described could work out okay.

**Comment by wholmgren** [comment]
Ok. It's hard for me to know the pros/cons of 1 or 2 functions at this point, so no objection here to making a second one. @alorenzo175 any additional comments before @dplarson dives into this?

I was wondering about the speed of your time calculation vs.`fx_index.time` (where pandas has to create a bunch of python objects). I had a ~100k index on hand and ran the following benchmarks:

```
%%timeit

df = data_dict[1444]
times = df.index.time

31.1 ms ± 613 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


%%timeit

​df = data_dict[1444]
times = df.index.hour + df.index.minute / 60

9.27 ms ± 28.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Not surprising, but also probably not a meaningful difference for our data.

**Comment by dplarson** [comment]
> Not surprising, but also probably not a meaningful difference for our data.

Thanks for this. It's always nice have a quantitative speed reference at hand when discussing.

**Comment by alorenzo175** [comment]
I don't think I have much to add at this point. I'll need to see the implementation to understand if one function or two or whatever option is appropriate.

**Comment by dplarson** [comment]
Sounds good. I'll add initial implementations of the two approaches (with some tests) and then we can discuss more before I move onto final polishing.

**Comment by dplarson** [comment]
As of commit 2405511, `persistence_probabilistic()` does the following:
- computes an empirical CDF using all data from `data_start` to `data_end`, without matching by time of day
- returns forecast probabilities (i.e. only supports providing `constant_values` with physical units)

**Comment by dplarson** [comment]
Apologies for the delay, but I got sidetracked by some other projects. I've got a good idea how to finish up this PR, but there are two areas I wanted to get feedback on:


#### 1) matching by time of day
Should this happen automatically within the function based on the observation data available (e.g. `if data_end - data_start > pd.TimeDelta("1D"): match_by_timeofday; else:  dont_match`) or instead be exposed as a user-selected parameter (e.g. `timeofday=True`: match by time of day)? Either way there would be checks to ensure there is "sufficient" data for matching by time of day (e.g. forecast 9:00am using the observations of 9:00am from the past 7 days).

#### 2) function output
Is it fine to keep the function as returning a list of `pandas.Series()` objects (one per `constant_value`) or is there a need to group together the `pandas.Series()` objects with corresponding `datamodel.ProbabilisticForecastConstantValues()` objects?

@wholmgren Thoughts? We can also discuss during the team call today.

**Comment by dplarson** [comment]
Note to self: based on the discussion yesterday, I will implement the code as follows:
1. split the code into two functions: one for using all data between `data_start` and `data_end` (with no matching for time of day) and another for matching by time of day
2. stick with returning the forecasts solely as a list of `pandas.Series()` objects

I'm working through the code now.

**Comment by dplarson** [comment]
I still need to add some additional explanation (with equations) to the docstrings, but this PR is effectively ready for review:
- `persistence_probabilistic()`: probabilistic persistence that considers all observations between `data_start` and `data_end`
- `persistence_probabilistic_timeofday()`: same as `persistence_probabilistic()` except the forecasts are matched by the time of day (e.g. if forecasting 9am, use an empirical CDF computed based solely on observations from 9am from previous days)

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 1]
```diff
@@ -362,3 +365,196 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    if axis == "x":
+        cdf = ECDF(obs)
+        forecasts = []
+        for constant_value in constant_values:
+            fx_prob = cdf(constant_value) * 100.0
+            forecasts.append(pd.Series(fx_prob, index=fx_index))
+    elif axis == "y":   # constant_values=percentiles, fx=variable
+        for constant_value in constant_values:
+            assert 0 <= constant_value <= 100, ("Constant percentile values "
+                                                "must be between 0 and 100")
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: I've removed this assert.

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 2]
```diff
@@ -362,3 +365,196 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+                             freq=interval_length, closed=closed)
+
+    obs = load_data(observation, data_start, data_end)
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
+
+    # confirm sufficient data for matching by time of day
+    assert data_end - data_start > pd.Timedelta('7D'), (
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Revised to use `raises ValueError`.

**Comment by dplarson** [comment]
> @dplarson it looks good. The only big thing is that these functions can't assume that observation and forecast interval label and interval length are consistent. It like the code would fail without some resampling.

Good point. I'll try revising to be robust to mistmatched observation and forecast `interval_label` and `interval_length`.

**Comment by dplarson** [comment]
I pushed a commit (e4a1443) that adds observation resampling (to match the forecast sampling) to only `persistence_probabilistic_timeofday()`. The resampling should not be needed for `persistence_probabilistic()` since it uses the entire observation dataset (`data_start` to `data_end`) to compute the empirical CDF or percentiles.

**Comment by wholmgren** [comment]
> The resampling should not be needed for persistence_probabilistic() since it uses the entire observation dataset (data_start to data_end) to compute the empirical CDF or percentiles.

I'm not sure about that. Say I have 1 minute SURFRAD observations but I want probabilistic persistence for 1 hr interval average. The statistics of the hourly average data is different than the 1 minute data.

**Comment by dplarson** [comment]
I see your point and I agree. I'll revise so both functions resample.

**Comment by dplarson** [comment]
@wholmgren In case it wasn't clear: this PR is ready for a full review.

**Comment by wholmgren** [comment]
Also, @dplarson what do you think about adding a clear sky index option like @kdayday discussed in [Benchmark probabilistic solar forecasts: Characteristics and recommendations](https://www.sciencedirect.com/science/article/pii/S0038092X20305429)? I really like it, but @kdayday concluded by recommending the complete history persistence ensemble as a default choice, so maybe save the index implementation for future work? That also reminds me that we need to add the Yang and Doubleday references to these functions.

**Comment by dplarson** [comment]
I think adding a clear sky index option makes a lot of sense, but I would lean towards adding that in a follow-up issue/PR, unless you had a strong preference for including it here/now.

Regardless, I'll add references of the Yang and Doubleday papers (which I agree make sense to cite).

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 3]
```diff
@@ -362,3 +365,220 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+
+
+def persistence_probabilistic(observation, data_start, data_end,
+                              forecast_start, forecast_end,
+                              interval_length, interval_label,
+                              load_data, axis, constant_values):
+    r"""
+    Make a probabilistic persistence forecast using the *observation* from
+    *data_start* to *data_end*. In the forecast literature, this method is
+    typically referred to as Persistence Ensemble (PeEn). [1]
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Revised.

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 4]
```diff
@@ -362,3 +365,220 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+
+    The function handles forecasting either constant variable values or
+    constant percentiles. In the examples below, we use GHI to be concrete but
+    the concepts also apply to other variables (AC power, net load, etc.).
+
+    If forecasting constant values (e.g. forecast the probability of GHI being
+    less than or equal to 500 W/m^2), the persistence forecast is:
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Revised as suggested.

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 5]
```diff
@@ -362,3 +365,220 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+        for constant_value in constant_values:
+            fx_prob = cdf(constant_value) * 100.0
+            forecasts.append(pd.Series(fx_prob, index=fx_index))
+    elif axis == "y":   # constant_values=percentiles, fx=variable
+        forecasts = []
+        for constant_value in constant_values:
+            fx = np.percentile(obs, constant_value)
+            forecasts.append(pd.Series(fx, index=fx_index))
+    else:
+        raise ValueError(f"Invalid axis parameter: {axis}")
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:479]: Add a `Raises` section to docstring for both functions.

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 6]
```diff
@@ -362,3 +365,220 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
+
+    # confirm sufficient data for matching by time of day
+    if data_end - data_start < pd.Timedelta('7D'):
+        raise ValueError("Insufficient data to match by time of day")
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:574]: The 7 day limit was a (somewhat) arbitrary choice by me when developing the code. I'll double-check the limit from the paper and use that. (Updating the tests as needed.)

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 7]
```diff
@@ -362,3 +365,220 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+
+    if axis == "x":
+        forecasts = []
+        for constant_value in constant_values:
+            fx = np.empty(len(fx_index))
```
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:579]: No particular reason. I like your suggestion and will revise accordingly.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Update: the Allesandrini et al. (2015) paper mentions 20 days, which is also discussed in the Doubleday et al. (2020) paper. So I switched to 20 days as the minimum and updated the tests.

**Comment by dplarson** [comment]
@alorenzo175 Can you take a look at this PR?

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 8]
```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    interval_length : pd.Timedelta
+        Forecast interval length
+    interval_label : str
+        instant, beginning, or ending
+    load_data : function
+        A function that loads the observation data. Must have the
+        signature load_data(observation, data_start, data_end) and
+        properly account for observation interval label.
+    axis : {'x', 'y'}
+        The axis on whicht he constant values of the CDF is specified. The axis
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/persistence.py:433]:
```suggestion
        The axis on which the constant values of the CDF is specified. The axis
```

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 9]
```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/persistence.py:525]:
```suggestion
        The axis on which the constant values of the CDF is specified. The axis
```

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 10]
```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    # observation data resampled to match forecast sampling
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
+
+    # confirm sufficient data for matching by time of day
+    if data_end - data_start < pd.Timedelta('20D'):
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/persistence.py:585]: hmm, should we actually compare 
```python
obs.last_valid_index() - obs.first_valid_index() < pd.Timedelta('20D')
```

``load_data`` likely only fetches the available data between start and end but doesn't guarantee there is data at all

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 11]
```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    fx_index = pd.date_range(start=forecast_start, end=forecast_end,
+                             freq=interval_length, closed=closed)
+
+    # observation data resampled to match forecast sampling
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/persistence.py:582]: need to be careful here about timezone. If somehow forecast_start/end have a different timezone from obs, matching by time of day might use the wrong data. 

```suggestion
    if fx_index.tzinfo is None:
        fx_index = fx_index.tz_localize(obs.index.tzinfo)
    else:
        fx_index = fx_index.tz_convert(obs.index.tzinfo)
    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
```

File: solarforecastarbiter/reference_forecasts/persistence.py [hunk 12]
```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
...
+    elif axis == "y":
+        forecasts = []
+        for constant_value in constant_values:
+            fx = pd.Series(np.nan, index=fx_index)
+            for tod in fx_timeofday:
+                data = obs[obs_timeofday == tod]
+                fx[fx_timeofday == tod] = np.percentile(data, constant_value)
+            forecasts.append(fx)
+    else:
+        raise ValueError(f"Invalid axis parameter: {axis}")
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/persistence.py:606]: missing from coverage. Add a simple test that this is raised

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 8 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:433]: Thanks for spotting this. I've corrected both.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 9 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Revised.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 10 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py]: Revised as suggested.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 11 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:586]: Revised as suggested.

Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:586]: I'll add a test for code coverage

Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:586]: I've added a test for this.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 12 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:610]: Added a test.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 11 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:586]: @alorenzo175 @wholmgren Actually, do we actually want the forecasts returned with the same timezone as the observations? Does that match the logic/expectations in the other reference forecasts? I don't see any `tz_convert` operations in `persistence.py`.

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/persistence.py:586]: I don't think it matters much, but yes, to be consistent with ``persistence.py`` maybe convert the observations to the forecast index timezone if not none, otherwise use the observation timezone

Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:584]: Sounds good. I've removed the `tz_convert` and just kept the `tz_localize` for if the fx is missing tz info.

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/persistence.py:584]: I think you still need a `tz_convert` for obs. For example, if the say `forecast_start` is specified as a PST timestamp, but the observations are returned as UTC. Then the `fx_index` will be PST but `obs.index` will be UTC and the `obs_timeofday` and `fx_timeofday` will be shifted. I think something like this will take care of it
```python
if obs.index.tzinfo is not None:
   obs = obs.tz_convert(fx_index.tzinfo)
else:
   obs = obs.tz_localize(fx_index.tzinfo)
```
which needs to go after fx_index is localized but before the obs_timeofday. The other persistence functions don't have a `tz_convert` because they aren't comparing time of day, although they might have issues if something is unlocalized.

**Comment by dplarson** [comment]
Just realized the code/tests may not cover an edge case properly: for the `persistence_probabilistic_timeofday()`, we compute a `timeofday` index for matching. But I'm not 100% sure that handles the case where the fx and obs have different timezones. I still think the fx should be returned with the same timezone as the `forecast_start` and `forecast_end` inputs, but I need to check that the `timeofday` matching correctly handles different timezones.

File: solarforecastarbiter/reference_forecasts/persistence.py (hunk 11 above)
Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:584]: You're correct. (I just wrote a comment saying something similar before I saw your reply.)

Comment (dplarson) [reply solarforecastarbiter/reference_forecasts/persistence.py:591]: With e4dd65f, the `persistence_probabilistic_timeofday()` should now properly handle all combinations of obs and fx timezones.

**Comment by alorenzo175** [comment]
thanks @dplarson
